api/models/User.js

```diff
@@ -31,4 +40,100 @@ userSchema.statics.create = function(name) {
 	});
 };
 
+userSchema.statics.read = function(GoogleID)
+{
+	return new Promise((resolve , reject) => {
+		mongoose.model('User', userSchema).findOne({googleID: GoogleID}, (err,user)=>{
+			if(err) reject(err);
+			else resolve(user);
+		});
+	});
+}
+
+userSchema.statics.delete = function(GoogleID)
+{
+	return new Promise((resolve, reject)=>{
+		mongoose.model('User', userSchema).findOne({googleID: GoogleID}).remove((err,offer)=>{
+			if(err) reject(err);
+		});
+	});
+}
+
+
+// Static: User.doSomething
+// Instance method: 
+// var user;
+// user.doSomething()
+
+
+userSchema.methods.setisAdminTo = function(becomeAdmin)
+{
+	this.isAdmin = becomeAdmin;
+	console.log('set user name:' + this.name + ' isAdmin to ' + becomeAdmin);
+	
+	return new Promise((resolve, reject)=>{
+		this.save((error, changedUser) => {
+			if(error) reject(error);
+			else resolve(changedUser);
+			});
+		});
+}
+
+
+userSchema.methods.setisMentorTo = function(becomeMentor)
+{
+	this.isMentor = becomeMentor;
+	console.log('set user name: '+this.name+' isMentor to '+becomeMentor);
+
+	return new Promise((resolve,reject)=>{
+		this.save((err, changedUser)=>{
+			if(err) reject(err);
+			else resolve(changedUser);
+		});
+	});
+}
+
+userSchema.methods.addSkill = function(skill)
+{
+	//check whether skill exists, should i throw err, if yes how?
+	var exists = false;
+	for(var i = 0; i < this.skills.length; i++)
+	{	
+		if(this.skills[i]===skill){
+			exists = true;
+			break;
+		}
+	}
+	if(!exists){
+		this.skills.push(skill);
+		console.log('skill: '+skill+' added')
+	}
+
+	return new Promise((resolve, reject)=>{
+		this.save((err,skillfulUser)=>{
+			if(err) reject(err);
+			else resolve(skillfulUser);
+		});
+	});
+}
+
+userSchema.methods.removeSkill = function(skill)
```

Comment:
There may be a mongoose built in method to remove something from and array in the DB. Try finding it instead of implementing it yourself.